PEER-204 Endpoint for Registration

sembcorpp opened the pull request:
Managed to add the endpoints for user registrations:
<img width="1091" alt="Screenshot 2024-09-23 at 1 38 23 AM" src="https://github.com/user-attachments/assets/3e612b25-1c31-4c16-89f3-03da5ae4c0d2">
<img width="1397" alt="Screenshot 2024-09-23 at 1 42 01 AM" src="https://github.com/user-attachments/assets/13350679-609d-483d-83e0-5a4896d12551">

sembcorpp committed:
adding user registration end points

sembcorpp committed:
Merge branch 'main' into PEER-204-Endpoint-For-Registration

syncing changes

sembcorpp committed:
clean up

SeeuSim commented on backend/user/src/server.ts:
```diff
@@ -12,6 +13,7 @@ app.use(pino());
 app.use(json());
 app.use(helmet());
 app.use('/auth', authRoutes);
+app.use('/register', registerRoutes);
```
@sembcorpp might wanna discuss with @TabrizPlv if this is the routing to go with, or to merge with auth routes

SeeuSim commented on backend/user/src/services/registration/register-logic.ts:
```diff
@@ -0,0 +1,60 @@
+  return {
+    code: StatusCodes.CREATED,
+    data: {
+      user: newUser,
```
Do not return the hashed password over REST

sembcorpp committed:
removed hashed password from REST

sembcorpp committed:
changed registration endpoint to be under auth

sembcorpp commented:
removed the hashed passwords from return output and changed the endpoint for register to be under auth

<img width="1138" alt="Screenshot 2024-09-23 at 11 04 36 AM" src="https://github.com/user-attachments/assets/c6c749c8-cbce-43c3-a188-15eea06f3eb2">

SeeuSim reviewed (approved):
lgtm, please merge

sembcorpp committed:
refactor register controller and service into auth

sembcorpp committed:
refactor index.ts naming to auth-controller.ts for better readability
